R/utils.R

```diff
@@ -3,14 +3,21 @@ can_overwrite <- function(path) {
 
   if (!file.exists(path)) {
     TRUE
-  } else if (interactive() && !yesno("Overwrite `", name, "`?")) {
+  } else if (interactive() && !nope("Overwrite `", name, "`?")) {
     TRUE
   } else {
     FALSE
   }
 }
 
-yesno <- function(...) {
+## TRUE: user selected one of the "no's" or selected nothing or not interactive
+##   or in tests
+## FALSE: user selected one of the "yes's"
+nope <- function(...) {
+  if (!interactive() || is_testing()) {
```

Comment:
Should it still display the message somehow? Should non-interactive usage be an error? 

The three uses cases above each handle non-interactive sessions in a different way. Maybe `nope()` should just do `stop("`nope()` called in non-interactive session")` to force developer to deal with it?

Reply:
I've made non-interactive usage an error. I haven't rationalized the functions wrt interactivity yet but will.